### ISSUE TITLE
messageRoom and send methods should return what adapter.send returns

### ISSUE THREAD

**Issue by xob** [hubotio/hubot#1498]
An adapter's `send` method could return something, and it would be nice to have access to it when using the `robot.messageRoom` and `robot.send` commands.

One example is hubot-slack, which returns an array of promises that one can use to get a reference to the messages sent, allowing the developer to then edit the message by using update commands.

Version 2.x of hubot allowed this, as it was written in coffeescript and therefore implicitely returned the what the last line of code in a function would return. At our company, the hubot code used that to edit slack messages to show server deployment progress without spamming messages. Verion 3 no longer allows that, as it was re-written in javascript and there is no `return` statement.

**Comment by vpeltola** [comment]
Noticed this problem as well when trying to upgrade from 2.9.1 to 3.3.2.  In 2.9.1 I had used robot.send to send a message, capture the returned array or promises to get a handle to the message, and then post additional messages in a thread under the first message.  Now this no longer works.
As xob explained, an easy fix here to restore the old behaviour would be to add the "return" statement in the robot.reply function:
```
send (envelope/* , ...strings */) {
    const strings = [].slice.call(arguments, 1)

    return this.adapter.send.apply(this.adapter, [envelope].concat(strings))
  }
```
(and the same in robot.messageRoom)

**Comment by rajasibero** [comment]
> An adapter's `send` method could return something, and it would be nice to have access to it when using the `robot.messageRoom` and `robot.send` commands.
> 
> One example is hubot-slack, which returns an array of promises that one can use to get a reference to the messages sent, allowing the developer to then edit the message by using update commands.
> 
> Version 2.x of hubot allowed this, as it was written in coffeescript and therefore implicitely returned the what the last line of code in a function would return. At our company, the hubot code used that to edit slack messages to show server deployment progress without spamming messages. Verion 3 no longer allows that, as it was re-written in javascript and there is no `return` statement.

**Comment by cristina-9** [comment]
Hi @seratch! Any chance of including this fix in an upcoming release? We would like to upgrade to a newer version but we are blocked by this issue.

**Comment by cristina-9** [comment]
hi @yaananth
I recently saw you wrote [this ](https://github.blog/2021-12-01-using-chatops-to-help-actions-on-call-engineers/)blog post about ChatOps using Hubot. I am stuck on an older version of Hubot (2.9.x) because of this issue. I was wondering if you know if Hubot is still maintained? I'd love for this fix to be added so I can upgrade to the latest version.

**Comment by yaananth** [comment]
@cristina-9 i got an internal update on this 

> we not forking the engine, we use it. We have internal engine and would have to post updates to public engine at some point

There is an internal issue where we tracking the status decision on that, I can keep you posted here when we have an update there!
